core/network/impl/peer_manager_impl.hpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_NETWORK_PEERMANAGERIMPL
+#define KAGOME_NETWORK_PEERMANAGERIMPL
+
+#include "network/peer_manager.hpp"
+
+#include <libp2p/host/host.hpp>
+#include <libp2p/protocol/kademlia/kademlia.hpp>
+#include <libp2p/protocol/identify/identify.hpp>
+#include <memory>
+#include <queue>
+
+#include "application/app_state_manager.hpp"
+#include "application/chain_spec.hpp"
+#include "clock/clock.hpp"
+#include "common/logger.hpp"
+#include "network/impl/stream_engine.hpp"
+#include "network/types/bootstrap_nodes.hpp"
+#include "network/types/own_peer_info.hpp"
+
+namespace kagome::network {
+
+  class PeerManagerImpl : public PeerManager,
+                          public std::enable_shared_from_this<PeerManagerImpl> {
+   public:
+    PeerManagerImpl(
+        std::shared_ptr<application::AppStateManager> app_state_manager,
+        std::shared_ptr<libp2p::Host> host,
+        std::shared_ptr<libp2p::protocol::Identify> identify,
+        std::shared_ptr<libp2p::protocol::kademlia::Kademlia> kademlia,
+        std::shared_ptr<libp2p::protocol::Scheduler> scheduler,
+        std::shared_ptr<StreamEngine> stream_engine,
+        std::shared_ptr<application::ChainSpec> config,
+        const clock::SteadyClock &clock,
+        const BootstrapNodes &bootstrap_nodes,
+        const OwnPeerInfo &own_peer_info);
+
+    /** @see AppStateManager::takeControl */
+    bool prepare();
+
+    /** @see AppStateManager::takeControl */
+    bool start();
+
+    /** @see AppStateManager::takeControl */
+    void stop();
+
+    /** @see PeerManager::connectToPeer */
+    void connectToPeer(const PeerInfo &peer_info) override;
+
+    /** @see PeerManager::forEachPeer */
+    void forEachPeer(std::function<void(const PeerId &)> func) const override;
+
+    /** @see PeerManager::forOnePeer */
+    void forOnePeer(const PeerId &peer_id,
+                    std::function<void()> func) const override;
+
+    /** @see PeerManager::forOnePeer */
+    void keepAlive(const PeerId &peer_id) override;
+
+   private:
+    /// Announce about himself ondemand
+    void announce();
+
+    /// Discovery other peers ondemand
+    void discovery();
+
+    /// Aligns cinnected stream amount
```

Comment:
```suggestion
    /// Aligns connected stream amount
```
Also, a bit unclear.